### PR TITLE
[DOCS] Update docs about linear tree and monotone constraints

### PR DESCRIPTION
**Pull request by btrotta**
As suggested by @flippercy https://github.com/microsoft/LightGBM/issues/3910#issuecomment-775502268

### CODE REVIEW

File: docs/Parameters.rst [hunk 1]
```diff
@@ -139,6 +139,8 @@ Core Parameters
 
       -  **Note**: setting ``linear_tree=true`` significantly increases the memory use of LightGBM
 
+      -  **Note**: if you specify ``monotone_constraints``, constraints will be enforced when choosing the split points, but not when fitting the linear models on leaves.
```
Comment (StrikerRUS) [inline docs/Parameters.rst:142]:
```suggestion
      -  **Note**: if you specify ``monotone_constraints``, constraints will be enforced when choosing the split points, but not when fitting the linear models on leaves
```

File: include/LightGBM/config.h [hunk 2]
```diff
@@ -158,6 +158,7 @@ struct Config {
   // descl2 = **Note**: only works with CPU and ``serial`` tree learner
   // descl2 = **Note**: ``regression_l1`` objective is not supported with linear tree boosting
   // descl2 = **Note**: setting ``linear_tree=true`` significantly increases the memory use of LightGBM
+  // descl2 = **Note**: if you specify ``monotone_constraints``, constraints will be enforced when choosing the split points, but not when fitting the linear models on leaves.
```
Comment (StrikerRUS) [inline include/LightGBM/config.h:161]:
```suggestion
  // descl2 = **Note**: if you specify ``monotone_constraints``, constraints will be enforced when choosing the split points, but not when fitting the linear models on leaves
```

**Review by StrikerRUS — APPROVED** [review]
Thanks a lot!
Just one nit for the consistency with the rest of the document.

**Review by jameslamb — APPROVED** [review]
thanks!

**Pull request merged**